### PR TITLE
CLN: update pandas.lib deprecation messages (GH15936)

### LINKED ISSUE #15936: Deprecated pandas.lib redirect messages
Currently, you get the following message when you try to access one of the `pandas.lib` functions:

```
In [1]: pd.lib.Timedelta
/home/joris/miniconda3/envs/dev/bin/ipython:1: FutureWarning: pandas.lib.Timedelta is deprecated. 
Please use pandas._libs.lib.Timedelta instead.
  #!/home/joris/miniconda3/envs/dev/bin/python
Out[1]: pandas._libs.tslib.Timedelta

In [1]: pd.lib.is_bool
/home/joris/miniconda3/envs/dev/bin/ipython:1: FutureWarning: pandas.lib.is_bool is deprecated. 
Please use pandas._libs.lib.is_bool instead.
  #!/home/joris/miniconda3/envs/dev/bin/python
Out[1]: <function pandas._libs.lib.is_bool>
```

(apart from `infer_dtypes`, for which there is already another redirect).

We shouldn't recommend to use `pandas._lib.lib`. 
For some there is a public alternative (like `Timestamp`, `Timedelta`), others are probably just private and shouldn't have an alternative. 

We could list all of them as `removals` and `moved` in `lib = _DeprecatedModule(..)`, but that will be a long list. Or we could just don't provide an alternative import path by default, and only include some specific ones for which there is an alternative.

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Closes #15936

### CODE REVIEW

File: pandas/__init__.py [hunk 1]
```diff
@@ -74,8 +74,11 @@
 parser = _DeprecatedModule(deprmod='pandas.parser',
                            removals=['na_values'],
                            moved={'CParserError': 'pandas.errors.ParserError'})
-lib = _DeprecatedModule(deprmod='pandas.lib', deprmodto='pandas._libs.lib',
-                        moved={'infer_dtype': 'pandas.api.lib.infer_dtype'})
+lib = _DeprecatedModule(deprmod='pandas.lib', deprmodto=False,
+                        moved={'Timestamp': 'pandas.Timestamp',
+                               'Timedelta': 'pandas.Timedelta',
+                               'NaT': 'pandas.NaT',
```
Comment (jreback) [inline pandas/__init__.py]: this is in .types now (i removed .lib)

Comment (jorisvandenbossche) [inline pandas/__init__.py]: Well, I first thought that as well, but it is not the case in master: https://github.com/pandas-dev/pandas/blob/master/pandas/api/lib/__init__.py

Comment (jreback) [inline pandas/__init__.py]: i didn't remove that file which is not distributed - look in types

Comment (jorisvandenbossche) [inline pandas/__init__.py]: I don't follow ..

It is also not in types: https://github.com/pandas-dev/pandas/blob/master/pandas/api/types/__init__.py

and why shouldn't the file be removed if we don't use it?

**Comment by jreback** [comment]
I *though* I had moved this (I may have changed some documents somewhere). Can you do it. remove ``api.lib`` and just put ``.infer_dtype`` into ``.api.types`` (and remove ``.lib``)

**Comment by jreback** [comment]
I'll do a separate PR to move infer_dtype

**Pull request merged**